### PR TITLE
Update gid API return codes.

### PR DESCRIPTION
**Pull request by hidmic**
To be compliant with the API spec. Connected to https://github.com/ros2/rmw/pull/274.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12383/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7352/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10095/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12296/)

File: rmw_connext_cpp/src/rmw_compare_gid_equals.cpp [hunk 1]
```diff
@@ -25,38 +25,22 @@ extern "C"
...
-    return RMW_RET_ERROR;
-  }
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  RMW_CHECK_ARGUMENT_FOR_NULL(result, RMW_RET_INVALID_ARGUMENT);
+
   auto detail1 = reinterpret_cast<const ConnextPublisherGID *>(gid1->data);
-  if (!detail1) {
-    RMW_SET_ERROR_MSG("gid1 is invalid");
-    return RMW_RET_ERROR;
-  }
```
Comment (clalancette) [inline rmw_connext_cpp/src/rmw_compare_gid_equals.cpp:54]: So can it be the case that `gid1->data` is nullptr?  Or are we just taking the approach of elsewhere where nullptr is just a special case of "undefined pointer"?

**Review by clalancette — COMMENTED** [review]
One question.

**Review by Lobotuerk — APPROVED** [review]
LGTM

File: rmw_connext_cpp/src/rmw_compare_gid_equals.cpp (hunk 1 above)
Comment (hidmic) [reply rmw_connext_cpp/src/rmw_compare_gid_equals.cpp:54]: `gid1->data` won't ever be `nullptr` -- [it's a primitive array](https://github.com/ros2/rmw/blob/6d3f68af511bb17dfdc9eef3dc345469e8042162/rmw/include/rmw/types.h#L475).

**Review by clalancette — APPROVED** [review]

**Pull request merged**